src/Server/middleware/bootstrapSharifyAndContextLocalsMiddleware.ts

```diff
@@ -84,3 +84,8 @@ export function updateContext(key: string, value: any) {
   const asyncLocalStorage = getAsyncLocalStorage()
   asyncLocalStorage.getStore()?.set(key, value)
 }
+
+export function getContext(key: string): any {
+  const asyncLocalStorage = getAsyncLocalStorage()
+  return asyncLocalStorage.getStore()?.get(key)
+}
```

Comment:
this asyncLocalStorage is impressive

Reply:
Yea - I avoided it at first, but after spending some time trying to thread things down from `SystemContext` and `Boot`, to try to get it to the Relay network layer middleware (from `createRelayEnvironment`) - I figured to try it out and it works great.

I suppose it can get out of hand, but it is a recommended way for request-specific stuff on the server, and on the client it's just an in-memory object.

It might be more, if there's an idiomatic way to do something more Relay-like - we should always prefer that - but something that's a bit outside that/atypical, it's probably cool to use it.

Reply:
I love this approach 💟

Reply:
Our wrapper is pretty clever - https://github.com/artsy/force/blob/fd32b1dd7348f47ac901a668a00649f80c3bcc83/src/Server/asyncLocalWrapper/index.ts

On the server, it uses the blessed `AsyncLocalStorage`, and on the client it just uses an in-memory `Map` object which it exposes via a `getStore` 'stub', so you can just use it agnostically - https://github.com/artsy/force/blob/fd32b1dd7348f47ac901a668a00649f80c3bcc83/src/Server/asyncLocalWrapper/browser.ts#L3-L8